File: 217_Contains_Duplicate.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
using namespace std;


class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> seen;
        for (int num: nums){
            if (seen[num]>=1){
                return true;
            }
            seen[num]++;
        }
        return false;
    }
};

int main(){
    Solution sol;
    vector<int> nums = {1, 2, 3, 4};
    bool ans = sol.containsDuplicate(nums);
    cout << ans << endl;
}

File: 217_contains_duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        seen = set(nums)
        if len(seen) == len(nums):
            return False
        else :
            return True

if __name__ == "__main__":
    sol = Solution()
    nums = [0, 1, 1, 2, 3]
    print(sol.containsDuplicate(nums))